src/arrayMethodSort.js

```diff
@@ -4,9 +4,25 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction = (a, b) => `${a}` > `${b}`) {
+    for (let i = 0; i < this.length - 1; i++) {
+      if (
+        compareFunction(this[i], this[i + 1]) === true ||
+        compareFunction(this[i], this[i + 1]) > 0
+      ) {
+        const temporary = this[i];
+
+        this[i] = this[i + 1];
+
+        this[i + 1] = temporary;
+
+        i = -1;
```

Comment:
Setting the index 'i' to -1 after a swap is a valid approach to restart the loop, but it's not efficient. Consider using a do-while loop and a swapped flag to reduce the number of iterations.